File: historyscan-back/src/test/java/io/sch/historyscan/domain/contexts/analysis/clocrevisions/filesystem/FileSystemNodeSerializerUtils.java
package io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.module.SimpleModule;
import io.sch.historyscan.common.JsonReader;

public final class FileSystemNodeSerializerUtils {
    private FileSystemNodeSerializerUtils() {
    }

    public static FileSystemNode serializeExpectedRoot(String expectedRootTestCase) throws JsonProcessingException {
        var expectedRootJson = JsonReader.toExpectedJson("filesystemtree", expectedRootTestCase);
        ObjectMapper mapper = new ObjectMapper();
        SimpleModule module = new SimpleModule();
        module.addDeserializer(FileSystemNode.class, new FileSystemNodeDeserializer());
        mapper.registerModule(module);
        return mapper.readValue(expectedRootJson, FileSystemNode.class);
    }
}

File: historyscan-back/src/test/kotlin/io/sch/historyscan/domain/contexts/analysis/clocrevisions/filesystem/FileSystemNodeSerializerUtils.kt
package io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.module.SimpleModule
import io.sch.historyscan.common.*

object FileSystemNodeSerializerUtils {
    @Throws(JsonProcessingException::class)
    fun serializeExpectedRoot(expectedRootTestCase: String): FileSystemNode? {
        val expectedRootJson = JsonReader.toExpectedJson("filesystemtree", expectedRootTestCase)
        val mapper = ObjectMapper()
        val module = SimpleModule()
        module.addDeserializer(FileSystemNode::class.java, FileSystemNodeDeserializer())
        mapper.registerModule(module)
        return mapper.readValue(expectedRootJson, FileSystemNode::class.java)
    }
}
